app/validators/taxon_validator.rb

```diff
@@ -1,6 +1,6 @@
 class TaxonValidator < ActiveModel::Validator
   def validate(edition)
-    if missing_taxons?(edition)
+    if missing_taxons?(edition) && edition.requires_taxon?
```

Comment:
Could you add a test for this, e.g. in `TaxonValidatorTest`.

Reply:
Alternatively, you could add the condition into the Edition model, e.g. `validates_with TaxonValidator, on: :publish, if: :requires_taxon?` then test it in `EditionTest`.

Reply:
If you go the `on: :publish` route, you have to call `invalid?(:publish)` in your test. Took me a while to figure that out 😓 

https://github.com/alphagov/whitehall/blob/9e7d05ec09af294ec3a07c642e3c976ce58afaf8/test/unit/app/models/corporate_information_page_test.rb#L617

Reply:
Done 👍